common-utils/src/common/deployment/__init__.py

```diff
@@ -356,125 +378,270 @@ class FlowSpec(BaseModel):
     """This is the object we use to deploy a flow."""
 
     flow: Flow = Field(..., description="Flow funciton object.")
-    secrets: list[FlowSecret] = Field(
-        [], description="List of FlowSecret objects to pass to task definition."
+    secrets: list[FlowEnvar] = Field(
+        [],
+        description=(
+            "List of secrets as FlowEnvar objects to pass to task definition."
+            "These should only be the secret names and not full arn."
+        ),
     )
     docker_env: StrictStr = Field(
         ...,
-        description="docker environment name from pyproject.toml, which is the key name in '[tool.prefect.envs.base]'",
+        description="docker environment name from pyproject.toml, which is the key name in '[tool.prefect.envs.<>]'",
     )
     ephemeral_storage_gb: conint(gt=19, lt=201) = 20  # type: ignore
     deployments: list[FlowDeployment] = []
-    _slugified_flow_name: StrictStr = PrivateAttr()
-    _slugified_fn_name: StrictStr = PrivateAttr()
-    _project_name: StrictStr = PrivateAttr()
 
     class Config:
         arbitrary_types_allowed = True
 
     def __init__(self, **data) -> None:
-        """Set the private fields on intansiation.
-        DISABLE_FLOW_SPEC will be set to true to disable validation for ECS execution.
```

Comment:
removing the DISABLE_FLOW_SPEC envar logic since we not going to do any initial validation the depends on external files or libraries